jni/src/v1736/com_amazon_opendistroforelasticsearch_knn_index_v1736_KNNIndex.cpp

```diff
@@ -202,10 +199,8 @@ JNIEXPORT void JNICALL Java_com_amazon_opendistroforelasticsearch_knn_index_v173
         index->LoadIndex(indexString);
         env->ReleaseStringUTFChars(indexPath, indexString);
         has_exception_in_stack(env);
-        jclass indexClass = env->GetObjectClass(indexObject);
-        jmethodID setIndex = env->GetMethodID(indexClass, "setIndex", "(J)V");
-        env->CallVoidMethod(indexObject, setIndex, (jlong)index);
-        has_exception_in_stack(env);
+
+        return (jlong) index;
 
         // free up memory
```

Comment:
Oops, you're right - I haven't been able to get any of this code properly tested, intent of the PR was to send to Vamshi some changes to consider in debugging race conditions and he had offered to take a look. Next version of the PR fixes ordering here and also fixes a bad type signature in init()

Reply:
Fixed in https://github.com/opendistro-for-elasticsearch/k-NN/pull/24/commits/84eb45c0a0d560c85d622eecfe74dc9b644feb08

Reply:
> Additionally, would it be possible to either create an issue or add a description of the race condition and the failure that occurred because it?

I don't have details on the crash itself because I don't have access to the nodes which crashed. I added a "Motivation" section to the top of this PR to describe the circumstances that led to the crash.